src/compiler/checker.ts

```diff
@@ -13317,13 +13317,13 @@ namespace ts {
                         .expression;                             // x
                     const classSymbol = checkExpression(className).symbol;
                     if (classSymbol && classSymbol.members && (classSymbol.flags & SymbolFlags.Function)) {
-                        return getInferredClassType(classSymbol);
+                        return getFlowTypeOfReference(node, getInferredClassType(classSymbol));

```

Comment:
Can you find out if there’s a performance hit for this PR? I would expect this line especially to be costly, since there are probably a lot of `this.property` references in an OO codebase.

Reply:
Never mind, I looked the file locally, outside github’s 3 lines of context, and see that this is for JS.